error: borrow of packed field is unsafe and requires unsafe function or block (error E0133)
  --> $DIR/packed_sneaky-1.rs:20:1
   |
20 | #[pin_project] //~ ERROR borrow of packed field is unsafe and requires unsafe function or block
   | ^^^^^^^^^^^^^^
   |
note: lint level defined here
  --> $DIR/packed_sneaky-1.rs:20:1
   |
20 | #[pin_project] //~ ERROR borrow of packed field is unsafe and requires unsafe function or block
   | ^^^^^^^^^^^^^^
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
   = note: for more information, see issue #46043 <https://github.com/rust-lang/rust/issues/46043>
   = note: fields of packed structs might be misaligned: dereferencing a misaligned pointer or even just creating a misaligned reference is undefined behavior

error: aborting due to previous error

