error: unexpected token
 --> $DIR/invalid.rs:7:10
  |
7 |     #[pin()] //~ ERROR unexpected token
  |          ^

error: unexpected token
  --> $DIR/invalid.rs:12:18
   |
12 | struct B<T>(#[pin(foo)] T); //~ ERROR unexpected token
   |                  ^

error: unexpected token
  --> $DIR/invalid.rs:16:12
   |
16 |     A(#[pin(foo)] T), //~ ERROR unexpected token
   |            ^

error: unexpected token
  --> $DIR/invalid.rs:22:14
   |
22 |         #[pin(foo)] //~ ERROR unexpected token
   |              ^

error: expected identifier
  --> $DIR/invalid.rs:27:27
   |
27 | #[pin_project(UnsafeUnpin,,)] //~ ERROR expected identifier
   |                           ^

error: duplicate `UnsafeUnpin` argument
  --> $DIR/invalid.rs:33:28
   |
33 | #[pin_project(UnsafeUnpin, UnsafeUnpin)] //~ ERROR duplicate `UnsafeUnpin` argument
   |                            ^^^^^^^^^^^

error: duplicate `PinnedDrop` argument
  --> $DIR/invalid.rs:39:27
   |
39 | #[pin_project(PinnedDrop, PinnedDrop)] //~ ERROR duplicate `PinnedDrop` argument
   |                           ^^^^^^^^^^

error: duplicate `UnsafeUnpin` argument
  --> $DIR/invalid.rs:45:40
   |
45 | #[pin_project(PinnedDrop, UnsafeUnpin, UnsafeUnpin)] //~ ERROR duplicate `UnsafeUnpin` argument
   |                                        ^^^^^^^^^^^

error: duplicate `PinnedDrop` argument
  --> $DIR/invalid.rs:51:40
   |
51 | #[pin_project(PinnedDrop, UnsafeUnpin, PinnedDrop, PinnedDrop)] //~ ERROR duplicate `PinnedDrop` argument
   |                                        ^^^^^^^^^^

error: aborting due to 9 previous errors

