error[E0412]: cannot find type `UnpinStructFoo` in this scope
  --> $DIR/unpin_sneaky.rs:11:16
   |
11 | impl Unpin for UnpinStructFoo {} //~ ERROR E0412,E0321
   |                ^^^^^^^^^^^^^^ not found in this scope
help: possible candidate is found in another module, you can import it into scope
   |
3  | use crate::UnpinStructFoo;
   |

error[E0321]: cross-crate traits with a default impl, like `std::marker::Unpin`, can only be implemented for a struct/enum type, not `[type error]`
  --> $DIR/unpin_sneaky.rs:11:1
   |
11 | impl Unpin for UnpinStructFoo {} //~ ERROR E0412,E0321
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ can't implement cross-crate trait with a default impl for non-struct/enum type

error: aborting due to 2 previous errors

Some errors have detailed explanations: E0321, E0412.
For more information about an error, try `rustc --explain E0321`.
